src/main/java/com/microsoft/aad/msal4j/IAuthenticationResult.java

```diff
@@ -44,4 +44,15 @@ public interface IAuthenticationResult extends Serializable {
      * @return access token expiration date
      */
     java.util.Date expiresOnDate();
+
+    /**
+     * The source of the tokens in a result. Will simply be the word "cache" if retrieved from the in-memory token cache,
+     * otherwise it will be the actual authority used by the library for the token request
+     *
+     * In general an authority value here will match the authority configured in the client app or acquire token request,
+     * however there are certain cases, such as when using regional endpoints, where the library may use a different authority
+     *
+     * @return source of tokens in result
+     */
+    String tokenSource();
```

Comment:
Is this newly introduced as part of the regional api design doc?

Reply:
This wasn't part of the region design doc, it's just a little info field to give customers some better feedback. When troubleshooting this issue with a customer there was confusion about where exactly the error was happening because we didn't have an easy way to see exactly where the library was sending requests (the tokens don't contain any regional info, and we were caching the global info regardless). They pointed out [this field](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/b23bbce16567739e895185a0ffb7dcc25d55663e/src/client/Microsoft.Identity.Client/AuthenticationResult.cs#L162) in .NET's AuthenticationResult and asked if we had anything similar.

Reply:
Discussed this, and decided to not add this field to AuthenticationResult. Instead, if an app developer wants this information a new debug-level log was added to TokenRequestExecutor which will print out exactly where the request is going.